egs/aishell/s10/run.sh

```diff
@@ -0,0 +1,320 @@
+      --cmd $train_cmd 5000 data/train $lang $ali_dir $treedir
+fi
+
+if  [[ $stage -le 18 ]]; then
+  echo "creating phone language-model"
+  $train_cmd exp/chain/log/make_phone_lm.log \
+    chain-est-phone-lm \
+     "ark:gunzip -c $treedir/ali.*.gz | ali-to-phones $treedir/final.mdl ark:- ark:- |" \
+     exp/chain/phone_lm.fst || exit 1
+fi
+
+if [[ $stage -le 19 ]]; then
+  echo "creating denominator FST"
+  copy-transition-model $treedir/final.mdl exp/chain/0.trans_mdl
+  cp $treedir/tree exp/chain
+  $train_cmd exp/chain/log/make_den_fst.log \
+    chain-make-den-fst exp/chain/tree exp/chain/0.trans_mdl exp/chain/phone_lm.fst \
+       exp/chain/den.fst exp/chain/normalization.fst || exit 1
+fi
+
+# You should know how to calculate your model's left/right context **manually**
+model_left_context=12
+model_right_context=12
+egs_left_context=$[$model_left_context + 1]
+egs_right_context=$[$model_right_context + 1]
+frames_per_eg=150,110,90
+
+if [[ $stage -le 20 ]]; then
+  echo "generating egs"
+  steps/nnet3/chain/get_egs.sh \
+    --alignment-subsampling-factor 3 \
+    --cmd $train_cmd \
+    --cmvn-opts "--norm-means=false --norm-vars=false" \
+    --frame-subsampling-factor 3 \
+    --frames-overlap-per-eg 0 \
+    --frames-per-eg $frames_per_eg \
+    --frames-per-iter 1500000 \
+    --generate-egs-scp true \
+    --left-context $egs_left_context \
+    --left-context-initial -1 \
+    --left-tolerance 5 \
+    --right-context $egs_right_context \
+    --right-context-final -1 \
+    --right-tolerance 5 \
+    --srand 0 \
+    --stage -10 \
+    data/fbank_pitch/train \
+    exp/chain exp/tri3a_lats exp/chain/egs
+fi
+
+if [[ $stage -le 21 ]]; then
+  echo "merging egs"
+  mkdir -p exp/chain/merged_egs
+  num_egs=$(ls -1 exp/chain/egs/cegs*.ark | wc -l)
+
+  minibatch_size=128
+
+  run.pl --max-jobs-run $nj JOB=1:$num_egs exp/chain/merged_egs/log/merge_egs.JOB.log \
+    nnet3-chain-shuffle-egs ark:exp/chain/egs/cegs.JOB.ark ark:- \| \
+    nnet3-chain-merge-egs --minibatch-size=$minibatch_size ark:- \
+      ark,scp:exp/chain/merged_egs/cegs.JOB.ark,exp/chain/merged_egs/cegs.JOB.scp || exit 1
+
+  rm exp/chain/egs/cegs.*.ark
+fi
+
+if [[ $stage -le 22 ]]; then
+  # Note: it might appear that this $lang directory is mismatched, and it is as
+  # far as the 'topo' is concerned, but this script doesn't read the 'topo' from
+  # the lang directory.
+  local/mkgraph.sh --self-loop-scale 1.0 data/lang_test exp/chain exp/chain/graph
+fi
+
+num_epochs=10
+lr=2e-3
+
+feat_dim=$(cat exp/chain/egs/info/feat_dim)
+output_dim=$(cat exp/chain/egs/info/num_pdfs)
+
+hidden_dim=625
+kernel_size_list="1, 3, 3, 3, 3, 3" # comma separated list
+stride_list="1, 1, 3, 1, 1, 1" # comma separated list
+
+log_level=info # valid values: debug, info, warning
+
+if [[ $stage -le 24 ]]; then
+  echo "training..."
+
+  mkdir -p exp/chain/train/tensorboard
+  # train_checkpoint=./exp/exp/train/epoch-5.pt
```

Comment:
`train_checkpoint` can be retrieved from the pt files.

Reply:
Done.